### PR TITLE
Improving coupling error checking

### PR DESCRIPTION
**Pull request by andrsd**
1. error out when coupling scalar variables where field ones are expected
2. error out when coupling field variables where scalar ones are expected
3. report the name of the object where the error occured

### CODE REVIEW

File: framework/include/base/Coupleable.h [hunk 1]
```diff
@@ -344,6 +351,14 @@ class Coupleable
   /// This will always be zero because the default values for optionally coupled variables is always constant
   VariableSecond _default_second;
 
+  /**
+   * Check that the right kind of variable is being coupled in
+   *
+   * @param var_name The name of the coupled variable
+   * @param comp The component of the coupled variable
+   */
+  void checkVar(const std::string & var_name, unsigned int comp);
```
Comment (andrsd) [inline framework/include/base/Coupleable.h]: I managed to sneak in some [beer code](http://www.czechvar.com/) in here

Comment (permcody) [reply framework/include/base/Coupleable.h]: Haha! Nice you should have been a little more blatant and actually spelled it that way too!

**Review by permcody — APPROVED** [review]

File: framework/src/base/ScalarCoupleable.C [hunk 2]
```diff
@@ -160,17 +171,38 @@ ScalarCoupleable::coupledScalarValueOlder(const std::string & var_name, unsigned
...
 ScalarCoupleable::coupledScalarDotDu(const std::string & var_name, unsigned int comp)
 {
+  checkVar(var_name, comp);
   MooseVariableScalar * var = getScalarVar(var_name, comp);
   return var->duDotDu();
 }
 
+void
+ScalarCoupleable::checkVar(const std::string & var_name, unsigned int comp)
+{
```
Comment (cpgr) [inline framework/src/base/ScalarCoupleable.C]: I'm getting an unused parameter warning for `comp` here. I can't see that you are using it anywhere?

Comment (andrsd) [reply framework/src/base/ScalarCoupleable.C]: Right, I got that in debug mode as well and meant to either comment it out or remove it. Apparently I forgot to do so. I will fix that... Thanks!

**Pull request merged**